parse/ast.go

```diff
@@ -246,8 +246,10 @@ func (e *ExpressionComparison) Accept(v Visitor) any {
 type ComparisonOperator string
 
 const (
-	ComparisonOperatorEqual              ComparisonOperator = "="
-	ComparisonOperatorNotEqual           ComparisonOperator = "!="
+	// We use distinct for equality because it can handle nulls.
+	// see https://stackoverflow.com/questions/36508815/not-equal-and-null-in-postgres
+	ComparisonOperatorEqual              ComparisonOperator = "IS NOT DISTINCT FROM"
+	ComparisonOperatorNotEqual           ComparisonOperator = "IS DISTINCT FROM"
```

Comment:
What if they wanted the expected behavior of `=` and `!=`?  My first though about the linked but  is that the procedure was written incorrectly.

Comparison with null in sql has similar semantics to other languages, and if we break from that then others more familiar might make an issue with the exact opposite "bug"?
https://kwilteam.slack.com/archives/C05RZNFKNJD/p1705595383455259?thread_ts=1705594086.484569&cid=C05RZNFKNJD

Reply:
Yeah that is fair. I guess we can add in `is distinct` expressions to procedures

Reply:
Yeah agree with @jchappelow, it's different behavior and we should keep it same as postgres